portfolio/src/main/webapp/script.js

```diff
@@ -51,12 +52,24 @@ function addRandomDestination() {
 }
 
 
-/** 
+/**
  * adds response from servlet
  */
 
-function getGreeting() {
-    fetch('/data').then((response) => response.text()).then((greeting) => {
-        document.getElementById('greeting-container').innerHTML = greeting;
-    });
+function setGreetingsFromServer() {
+  fetch('/data').then((response) => response.json()).then((obj) => {
+    const greetingContainer = document.getElementById('greeting-container');
+
+    for (let i = 0; i < Object.keys(obj).length; i++) {
```

Comment:
Looking at your server and looking at this, do you need `Object.keys` here? Not a javascript expert but I'd expect you to be receiving a payload like `["Cindy", "Anthony", "Charles", "Chris", "Ben" ]` from the server. That string as json object should be an array right? In that case shouldn't it just be `obj.length`? There might be something safer about `Object.keys(obj).length` that's safer that I'm not aware of.

You can also do this functionally if you want:
```javascript
obj.map(createListElem).forEach(ele => greetingContainer.appendChild(ele));
```